app/uk/gov/hmrc/helptosave/models/account/Transactions.scala

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosave.models.account
+
+import java.time.LocalDate
+
+import cats.Eq
+import cats.data.Validated.{Invalid, Valid}
+import cats.data.{NonEmptyList, ValidatedNel}
+import cats.instances.list._
+import cats.instances.string._
+import cats.syntax.apply._
+import cats.syntax.eq._
+import cats.syntax.traverse._
+import play.api.libs.json.{JsString, JsValue, Json, Writes}
+
+sealed trait Operation
+case object Credit extends Operation
+case object Debit extends Operation
+
+object Operation {
+  implicit val eqOperation: Eq[Operation] = Eq.fromUniversalEquals
+  implicit val writes: Writes[Operation] = new Writes[Operation] {
+    override def writes(o: Operation): JsValue = JsString(o match {
+      case Credit ⇒ "credit"
+      case Debit  ⇒ "debit"
+    })
+  }
+}
+
+case class ValidNsiTransaction(
```

Comment:
This is only being used in the `apply` method in `Transactions`. To make it clearer that it's not for external use it'd be more obvious to make this case class unavailable outside of this file - perhaps as a private case class under in `Transactions`'s companion object

Reply:
it needs to be accessible by `Transaction.apply(vnt: ValidNsiTransaction, balanceAfter: BigDecimal)` so perhaps we should just make `ValidNsiTransaction` and its companion object `private` instead of moving them inside `Transactions`?

Reply:
sounds good

Reply:
done